### PR TITLE
[028,034]Test shirshov

### PR DESCRIPTION
**Pull request by ShirshovAlex**
add testContainers
add tests

### CODE REVIEW

**Review by Maksim-maevsky — DISMISSED** [review]

File: src/test/java/by/exadel/internship/formTest/FormTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+        String content = result.getResponse().getContentAsString();
+        FormFullDTO formFullDTO = objectMapper.readValue(content, FormFullDTO.class);
+
+        assertEquals(formFullDTO.getFirstName(), "testName");
+    }
+
+    @Test
+    public void checkListSize() throws Exception {
+        MvcResult result = mockMvc.perform(MockMvcRequestBuilders.get("/forms"))
```
Comment (mortal-gpt) [inline src/test/java/by/exadel/internship/formTest/FormTest.java]: Where did you create testData for this test? If you reuse data from previous test, it's wrong way. Tests should be independent.

File: src/test/java/by/exadel/internship/formTest/FormTest.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                .file(form)
+                .file(file)
+        )
+                .andExpect(status().isCreated())
+                .andReturn();
+
+        String content = result.getResponse().getContentAsString();
+        FormFullDTO formFullDTO = objectMapper.readValue(content, FormFullDTO.class);
+
+        assertEquals(formFullDTO.getFirstName(), "testName");
```
Comment (mortal-gpt) [inline src/test/java/by/exadel/internship/formTest/FormTest.java:48]: Are there other fields, that you can check?

File: src/test/java/by/exadel/internship/formTest/FormTest.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    @Test
+    public void givenFormWithFile_checkFirstName()
+            throws Exception {
+
+        MockMultipartFile file = new MockMultipartFile("file", "text.txt",
+                MediaType.MULTIPART_FORM_DATA_VALUE, "some file".getBytes());
+        MockMultipartFile form = new MockMultipartFile("form", "form",
+                MediaType.APPLICATION_JSON_VALUE,
+                ("{\"firstName\": \"testName\",\"city\": \"string\",\"country\": \"string\"," +
```
Comment (mortal-gpt) [inline src/test/java/by/exadel/internship/formTest/FormTest.java:32]: I think it's better to move test data to separate method or class.

File: src/test/java/by/exadel/internship/InternshipApplicationTests.java [hunk 4]
```diff
@@ -1,13 +1,71 @@
...
+    public void setUp() {
+        objectMapper.registerModule(new JavaTimeModule());
+    }
+
+    @BeforeEach
+    public void setup() throws Exception {
+        this.mockMvc = MockMvcBuilders.webAppContextSetup(this.webApplicationContext).build();
+    }
+
+    protected MvcResult getResult(HttpMethod method, URI uri, ResultMatcher status) throws Exception {
```
Comment (mortal-gpt) [inline src/test/java/by/exadel/internship/InternshipApplicationTests.java:53]: Use this method in tests.

File: src/test/java/by/exadel/internship/internshipTest/InternshipTest.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+                .andExpect(status().isOk())
+                .andReturn();
+
+        String content = result.getResponse().getContentAsString();
+
+        GuestInternshipDTO guestInternshipDTO = objectMapper
+                .readValue(content, new TypeReference<>() {
+                });
+
+        assertEquals(guestInternshipDTO.getName(), "some name");
```
Comment (mortal-gpt) [inline src/test/java/by/exadel/internship/internshipTest/InternshipTest.java:32]: check other fields, please

File: src/main/resources/application.yaml [hunk 6]
```diff
@@ -6,8 +6,8 @@ spring:
   datasource:
     driver-class-name: org.postgresql.Driver
     url: ${SPRING_DATASOURCE_URL}
-    password: ${SPRING_DATASOURSE_PASSWORD}
-    username: ${SPRING_DATASOURSE_USERNAME}
+    password: ${SPRING_DATASOURCE_PASSWORD}
+    username: ${SPRING_DATASOURCE_USERNAME}
```
Comment (kzaretskaya) [inline src/main/resources/application.yaml:10]: If you change it here, change these variables in docker-compose also.

File: src/test/java/by/exadel/internship/formTest/FormTest.java (hunk 1 above)
Comment (ShirshovAlex) [reply src/test/java/by/exadel/internship/formTest/FormTest.java]: we dont create testData in this test.

File: src/test/java/by/exadel/internship/formTest/FormTest.java (hunk 2 above)
Comment (ShirshovAlex) [reply src/test/java/by/exadel/internship/formTest/FormTest.java:48]: yes, done

File: src/test/java/by/exadel/internship/formTest/FormTest.java (hunk 3 above)
Comment (ShirshovAlex) [reply src/test/java/by/exadel/internship/formTest/FormTest.java:32]: this testData use only once, and only in this test

File: src/test/java/by/exadel/internship/InternshipApplicationTests.java (hunk 4 above)
Comment (ShirshovAlex) [reply src/test/java/by/exadel/internship/InternshipApplicationTests.java:53]: done

File: src/test/java/by/exadel/internship/internshipTest/InternshipTest.java (hunk 5 above)
Comment (ShirshovAlex) [reply src/test/java/by/exadel/internship/internshipTest/InternshipTest.java:32]: add fields, done

File: src/main/resources/application.yaml (hunk 6 above)
Comment (ShirshovAlex) [reply src/main/resources/application.yaml:10]: done

**Review by Maksim-maevsky — DISMISSED** [review]

File: src/main/java/by/exadel/internship/dto/internshipDTO/GuestInternshipDTO.java [hunk 7]
```diff
@@ -40,13 +40,13 @@
...
     @NotNull
-    private List<LocationDTO> countryList;
+    private List<LocationDTO> countryList;//+
 
     @NotNull
     private InternshipFormat internshipFormat;
 
     @NotNull
-    private Set<Skill> skills;
+    private Set<Skill> skills;//+
```
Comment (mortal-gpt) [inline src/main/java/by/exadel/internship/dto/internshipDTO/GuestInternshipDTO.java]: meaningless comment.

Comment (ShirshovAlex) [reply src/main/java/by/exadel/internship/dto/internshipDTO/GuestInternshipDTO.java]: done

**Review by Maksim-maevsky — APPROVED** [review]

**Review by viktarkolbik — APPROVED** [review]

**Pull request merged**